services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeYhteystietoEntity.kt

```diff
@@ -0,0 +1,45 @@
+package fi.hel.haitaton.hanke
+
+import java.time.LocalDateTime
+import javax.persistence.*
+
+enum class ContactType {
+    OMISTAJA, //owner
+    ARVIOIJA, //planner or person to do the planning of hanke
+    TOTEUTTAJA // implementator or builder
+}
+
+
+@Entity
+@Table(name = "hankeyhteystieto")
+class HankeYhteystietoEntity (
```

Comment:
All these fields are not really necessary inside constructor. Are any of them needed inside constructor? Maybe for testing some are but not all?

Reply:
Hmm.. this was how the example seemed to be written by Markku. If the instance is created usually it will contain these. 
Is there a good reason why NOT put them to constructor?

Reply:
Well.. if we want to create this kind of object, it would need all the fields (without default value) to be filled also. And these are Entity classes meaning that JPA/Hibernate will construct them without parameters and use setters to populate it (afaik). In principle, the constructor should only hold the fields that are mandatory in order to construct the object. If all these are mandatory, then it is okay. But maybe they are not all mandatory and some of them could be moved outside constructor? Those with default values for example?

Reply:
My previous example should not be used as a template; I had some issues while trying to create the initial entity, so it ended up like that, at the time. Indeed, probably many of the constructor parameters should be moved out to body also in HankeEntity.

Reply:
We don't have clear understanding of what is mandatory fields and what not :( I would maybe do this later if we have time... If there is no clear "you shall not do this because there will be massive bad consequences" rule why this should be changed immediately?